modal/dict.py

```diff
@@ -93,17 +94,49 @@ def persisted(
         stub.dict = Dict.from_name("my-dict")
         ```
         """
-        return _Dict.new()._persist(label, namespace, environment_name)
 
-    def persist(
-        self, label: str, namespace=api_pb2.DEPLOYMENT_NAMESPACE_WORKSPACE, environment_name: Optional[str] = None
+        async def _load(provider: _Dict, resolver: Resolver, existing_object_id: Optional[str]):
+            req = api_pb2.DictGetOrCreateRequest(
+                deployment_name=label,
+                namespace=namespace,
+                environment_name=_get_environment_name(environment_name),
+                object_creation_type=(api_pb2.OBJECT_CREATION_TYPE_CREATE_IF_MISSING if create_if_missing else None),
+            )
+            response = await resolver.client.stub.DictGetOrCreate(req)
+            logger.debug("Created dict with id %s" % response.dict_id)
+            provider._hydrate(response.dict_id, resolver.client, None)
+
+        return _Dict._from_loader(_load, "Dict()")
+
+    @staticmethod
+    def persisted(
```

Comment:
Do we plan to deprecate this at some point?

Reply:
Yes! But not quite yet

Reply:
I want to deprecate all `.persisted` methods once I've gone through all object classes